Xamarin.Forms.Platform.iOS/Renderers/LabelRenderer.cs

```diff
@@ -30,16 +30,31 @@ public override SizeRequest GetDesiredSize(double widthConstraint, double height
 				_perfectSizeValid = true;
 			}
 
-			if (widthConstraint >= _perfectSize.Request.Width && heightConstraint >= _perfectSize.Request.Height)
+			var widthFits = widthConstraint >= _perfectSize.Request.Width;
+			var heightFits = heightConstraint >= _perfectSize.Request.Height;
+
+			if (widthFits && heightFits)
 				return _perfectSize;
 
 			var result = base.GetDesiredSize(widthConstraint, heightConstraint);
-			result.Minimum = new Size(Math.Min(10, result.Request.Width), result.Request.Height);
-			if (Element.LineBreakMode != LineBreakMode.NoWrap)
+			var tinyWidth = Math.Min(10, result.Request.Width);
+			result.Minimum = new Size(tinyWidth, result.Request.Height);
+
+			if (widthFits || Element.LineBreakMode == LineBreakMode.NoWrap)
```

Comment:
This is the change that fixes this bug. Don't continue on to the "expand" logic if the text already fits the container.